File: ios/Swapl/Core/Repositories/StoryRepository.swift
import Foundation

// "Your Swapl story" networking (DOK-158). Thin relay over GET /api/me/story —
// the same endpoint the web client uses. The timeline, counts and referral code
// are all computed server-side from COMPLETED swaps and completed Keys stays;
// this layer only reads them.
final class StoryRepository: @unchecked Sendable {
    static let shared = StoryRepository()

    // GET /api/me/story — the caller's travel/hosting timeline, headline counts,
    // and shareable referral code. Bearer-authed; 401 when unauthenticated.
    func myStory() async throws -> SwaplStory {
        try await APIClient.shared.send("GET", "/api/me/story")
    }
}

File: android/swapl/app/src/main/java/app/swapl/core/repository/StoryRepository.kt
package app.swapl.core.repository

import app.swapl.core.model.SwaplStory
import app.swapl.core.network.ApiClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import javax.inject.Inject
import javax.inject.Singleton

// "Your Swapl story" networking (DOK-158). Thin relay over GET /api/me/story —
// the same endpoint the web/iOS clients use. All aggregation (timeline ordering,
// distinct city/country counts, the referral share block) lives server-side; the
// client only renders what it's handed.
@Singleton
class StoryRepository @Inject constructor(private val api: ApiClient) {

    // GET /api/me/story — the caller's travel/hosting timeline, headline counts,
    // and referral share block. Returns 401 when unauthenticated.
    suspend fun story(): SwaplStory =
        api.client.get("${api.baseUrl}/api/me/story").body()
}
